builtin/providers/gitlab/resource_gitlab_project.go

```diff
@@ -82,6 +90,7 @@ func resourceGitlabProjectSetToState(d *schema.ResourceData, project *gitlab.Pro
 	d.Set("snippets_enabled", project.SnippetsEnabled)
 	d.Set("visibility_level", visibilityLevelToString(project.VisibilityLevel))
 
+	d.Set("id", project.ID)
```

Comment:
@richardc it would be worth documenting then as I cannot see that in `gitlab_project` Attributes Reference